src/cdomains/floatDomain.ml

```diff
@@ -169,80 +180,95 @@ module FloatIntervalImpl(Float_t : CFloatType) = struct
   let starting s = of_interval' (Float_t.of_float Down s, Float_t.upper_bound)
   let starting_after s = of_interval' (Float_t.succ @@ Float_t.of_float Down s, Float_t.upper_bound)
 
-  let minimal x = Option.bind x (fun (l, _) -> Float_t.to_float l)
-  let maximal x = Option.bind x (fun (_, h) -> Float_t.to_float h)
+  let minimal = function
```

Comment:
Maybe we should even throw an exception here as well, because now `cast_to` creates a valid `Top` interval by calling `minimal/maximal` on `Bot`, which should probably fail.